lib/node_modules/@stdlib/types/index.d.ts

```diff
@@ -593,6 +593,85 @@ declare module '@stdlib/types/array' {
 	* const x: Collection<number> = [ 1, 2, 3 ];
 	*/
 	type Collection<T = any> = Array<T> | TypedArray | ArrayLike<T>; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `float64`: double-precision floating-point numbers (IEEE 754)
+	* -   `float32`: single-precision floating-point numbers (IEEE 754)
+	*/
+	type RealFloatingPointArrayMap = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
+		'float64': Float64Array;
+		'float32': Float32Array;
+	};
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `complex128`: double-precision complex floating-point numbers
+	* -   `complex64`: single-precision complex floating-point numbers
+	*/
+	type ComplexArrayMap = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
+		'complex64': Complex64Array;
+		'complex128': Complex128Array;
+	};
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `complex64`: single-precision complex floating-point numbers
+	* -   `int32`: 32-bit two's complement signed integers
+	* -   `uint32`: 32-bit unsigned integers
+	* -   `int16`: 16-bit two's complement signed integers
+	* -   `uint16`: 16-bit unsigned integers
+	* -   `int8`: 8-bit two's complement signed integers
+	* -   `uint8`: 8-bit unsigned integers
+	* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
+	*/
+	type IntegerArrayMap = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
```

Comment:
What is the disabled rule for? What is inconsistent?

Reply:
As configured, the [lint rule](https://typescript-eslint.io/rules/consistent-type-definitions/) enforces consistent use of interfaces instead of types. However, in this case we should use a type both for the composability and also since we don't want users extending these, so I had to disable the rule.